lib/util/url.ts

```diff
@@ -106,3 +109,43 @@ export function parseLinkHeader(
   }
   return _parseLinkHeader(linkHeader);
 }
+
+// this method returns path depth of a url
+// example : https://github.com/org/repo/nested-path
+// output : 3.  (org->repo->nested-path depth of this path is 3)
+export function urlPathDepth(url: string | undefined): number {
```

Comment:
I don't think we need this

Reply:
I have made examples on why this check is needed: 
https://github.com/renovatebot/renovate/blob/2bf2f3f4b763573c0bdcb5e06e90beb93f9981af/lib/modules/datasource/metadata.ts#L206-L214

So for these cases we do not need to keep `homepage` url.